core/trino-main/src/test/java/io/trino/operator/output/TestPartitionedOutputOperator.java

```diff
@@ -13,248 +13,693 @@
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.testng.Assert.assertEquals;
 
+@Test(singleThreaded = true)
 public class TestPartitionedOutputOperator
 {
-    private static final TrinoOperatorFactories TRINO_OPERATOR_FACTORIES = new TrinoOperatorFactories();
+    private static final OperatorFactories TRINO_OPERATOR_FACTORIES = new TrinoOperatorFactories();
+    private static final Session TEST_SESSION = testSessionBuilder().build();
     private static final DataSize MAX_MEMORY = DataSize.of(50, MEGABYTE);
     private static final DataSize PARTITION_MAX_MEMORY = DataSize.of(5, MEGABYTE);
 
-    private static final int PAGE_COUNT = 10;
-    private static final int POSITIONS_PER_PAGE = 1000;
-    private static final int PARTITION_COUNT = 512;
+    private static final int POSITIONS_PER_PAGE = 8;
+    private static final int PARTITION_COUNT = 2;
 
-    private static final List<Type> TYPES = ImmutableList.of(BIGINT);
-    private static final List<Type> REPLICATION_TYPES = ImmutableList.of(BIGINT, BIGINT);
-
-    private static final Block NULL_BLOCK = new RunLengthEncodedBlock(BIGINT.createBlockBuilder(null, 1).appendNull().build(), POSITIONS_PER_PAGE);
-    private static final Block TESTING_BLOCK = createLongSequenceBlock(0, POSITIONS_PER_PAGE);
-    private static final Block TESTING_DICTIONARY_BLOCK = createLongDictionaryBlock(0, POSITIONS_PER_PAGE);
-    private static final Block TESTING_RLE_BLOCK = createRLEBlock(new Random(0).nextLong(), POSITIONS_PER_PAGE);
-    private static final Page TESTING_PAGE = new Page(TESTING_BLOCK);
-    private static final Page TESTING_PAGE_WITH_NULL_BLOCK = new Page(POSITIONS_PER_PAGE, NULL_BLOCK, TESTING_BLOCK);
+    private static final PagesSerdeFactory PAGES_SERDE_FACTORY = new PagesSerdeFactory(createTestMetadataManager().getBlockEncodingSerde(), false);
+    private static final PagesSerde PAGES_SERDE = PAGES_SERDE_FACTORY.createPagesSerde();
 
+    private final Session testSession;
     private final OperatorFactories operatorFactories;
 
     private ExecutorService executor;
     private ScheduledExecutorService scheduledExecutor;
+    private TestOutputBuffer outputBuffer;
 
     public TestPartitionedOutputOperator()
     {
-        this(TRINO_OPERATOR_FACTORIES);
+        this(TEST_SESSION, TRINO_OPERATOR_FACTORIES);
     }
 
-    protected TestPartitionedOutputOperator(OperatorFactories operatorFactories)
+    protected TestPartitionedOutputOperator(Session testSession, OperatorFactories operatorFactories)
     {
+        this.testSession = testSession;
         this.operatorFactories = operatorFactories;
     }
 
     @BeforeClass
-    public void setUp()
+    public void setUpClass()
```

Comment:
undo

Reply:
I added `@BeforeMethod setUp` and it clashes with this name. I think `setUpClass` is a better name for  `@BeforeClass`